### PR TITLE
Pass both `fromAmount` and `toAmount` in authorization `IApplication` callbacks

### PR DESCRIPTION
**Pull request by pdyraga**
Refs https://github.com/keep-network/keep-core/issues/2715
See https://github.com/keep-network/sortition-pools/pull/143

Instead of passing just `toAmount` (previously named `amount`) we now
pass two values. This is helpful for applications that need to update
internal state. One example is an update of a sortition pool for
operator deauthorizing to 0: we want to remove them from the sortition
pool by setting their in-pool weight to 0 while still keeping them
earning rewards using their previous in-pool weight for the entire
duration of authorization decrease delay.

Gas cost impact feels negligible:
```
BEFORE
|  TokenStaking  ·  forceDecreaseAuthorization        ·          -  ·          -  ·      33697  │
|  TokenStaking  ·  increaseAuthorization             ·      48568  ·     129541  ·     122814  │
|  TokenStaking  ·  requestAuthorizationDecrease      ·      40358  ·      43182  ·      42403  │
|  TokenStaking  ·  requestAuthorizationDecrease      ·      44894  ·      63636  ·      55487  │

AFTER
|  TokenStaking  ·  forceDecreaseAuthorization        ·          -  ·          -  ·      33925  │
|  TokenStaking  ·  increaseAuthorization             ·      49079  ·     130052  ·     123325  │
|  TokenStaking  ·  requestAuthorizationDecrease      ·      40939  ·      43763  ·      42984  │
|  TokenStaking  ·  requestAuthorizationDecrease      ·      45475  ·      64795  ·      56395  │
```

### CODE REVIEW

**Comment by pdyraga** [comment]
@vzotova @cygnusv @eth-r This is a simple yet post-audit change and I would love to have two separate approvals before the merge 🙏🏻

**Review by vzotova — APPROVED** [review]
Looks good to me, ideally to add few lines to tests with checking that application receives proper `from` value

File: contracts/test/TokenStakingTestSet.sol
Comment (nkuba) [inline contracts/test/TokenStakingTestSet.sol:254]: Shouldn't the `toAmount` be less than the current authorization?
```suggestion
        require(toAmount < operatorStruct.authorized, "Nothing to decrease");
```

**Review by nkuba — APPROVED** [review]

**Pull request merged**
